First whack at containerizing ble-meshd

jcrossley3 opened the pull request:
I'm not entirely sure whether the gateway will be colocated with meshd
in the same container yet. Regardless, this is an attempt to run meshd
in a container...

...and it doesn't quite work yet.

Run the following:
```
  $ docker build -t drogue-gateway deploy/
  $ docker run --net=host --privileged -it drogue-gateway
```

jcrossley3 committed:
First whack at containerizing ble-meshd

I'm not entirely sure whether the gateway will be colocated with meshd
in the same container yet. Regardless, this is an attempt to run meshd
in a container...

...and it doesn't quite work yet.

Run the following:

  $ docker build -t drogue-gateway deploy/
  $ docker run --net=host --privileged -it drogue-gateway

dejanb commented:
Thanks Jim.

Yeah, I think we should have them collocated. Then running the gateway (with properly provisioned mesh) would be just a single command.

Some questions:

1. Should we try to install 5.6.x version of bluez, like we did in the tests?
2. Maybe for container we can start bluetooth-meshd as a service, rather than a process?

jcrossley3 commented:
> Yeah, I think we should have them collocated. Then running the gateway (with properly provisioned mesh) would be just a single command.

Cool. I like that.

> 1. Should we try to install 5.6.x version of bluez, like we did in the tests?

Yes, eventually. My immediate goal is to see the `mesh-cfgclient` provisioner recognize the mesh in the container. Makes me wonder if a 5.6 provisioner can talk to a 5.5 mesh, though?

> 2. Maybe for container we can start bluetooth-meshd as a service, rather than a process?

Yes, once I'm satisfied it works at all. :)

jcrossley3 committed:
Install newer version of bluez, 5.62

jcrossley3 committed:
Configure the image with keys from a provisioned mesh

We set things up per
https://gist.github.com/dejanb/85282932d08d26dce2c9323b3d0cc1e2

Only to realize that the scripts deriving from test-mesh still expect
to connect to the mesh via dbus, which isn't bridged from the
container.

Need to test with a real device and then service-fy meshd and add the
gateway script.

jcrossley3 commented:
Made some progress today after re-imaging my pi after losing my hci adapter somehow, but ultimately realized the scripts need dbus access to the mesh, which I don't plan to bridge into the container. Tomorrow, I'll test with a provisioned nRF which won't care about dbus. Once I know that works, I'll "service-fy" meshd within the controller and add the gateway.

jcrossley3 committed:
Service-ify bluetooth-meshd

We just copied the /etc/init.d/bluetooth script and changed its DAEMON
value. Still not sure how to deal with the meshd logs if'n we ever
need to.
